core/taskengine/vm_runner_contract_write.go

```diff
@@ -123,31 +115,63 @@ func (r *ContractWriteProcessor) Execute(stepID string, node *avsproto.ContractW
 				PaymasterAndData:     common.Bytes2Hex(userOp.PaymasterAndData),
 				Signature:            common.Bytes2Hex(userOp.Signature),
 			},
-
-			TxReceipt: &avsproto.Evm_TransactionReceipt{
-				Hash:        txReceipt.TxHash.Hex(),
-				BlockHash:   txReceipt.BlockHash.Hex(),
-				BlockNumber: uint64(txReceipt.BlockNumber.Int64()),
-				// TODO: Need to fetch this, it isn't available
-				//From:            txReceipt.From.Hex(),
-				//To:              txReceipt.To.Hex(),
-				GasUsed:           txReceipt.GasUsed,
-				GasPrice:          uint64(txReceipt.EffectiveGasPrice.Int64()),
-				CumulativeGasUsed: txReceipt.CumulativeGasUsed,
-				// Fee:                 txReceipt.Fee,
-				ContractAddress: txReceipt.ContractAddress.Hex(),
-				Index:           uint64(txReceipt.TransactionIndex),
-				// TODO: convert raw log
-				//Logs:                txReceipt.Logs,
-				LogsBloom:    common.Bytes2Hex(bloom),
-				Root:         common.Bytes2Hex(txReceipt.PostState),
-				Status:       uint32(txReceipt.Status),
-				Type:         uint32(txReceipt.Type),
-				BlobGasPrice: blobGasPrice,
-				BlobGasUsed:  uint64(txReceipt.BlobGasUsed),
-			},
 		},
 	}
+
+	// Only add TxReceipt if it exists
+	if txReceipt != nil {
+		var bloom []byte
+		bloom, _ = txReceipt.Bloom.MarshalText()
+
+		blobGasPrice := uint64(0)
+		if txReceipt.BlobGasPrice != nil {
+			blobGasPrice = uint64(txReceipt.BlobGasPrice.Int64())
+		}
+
+		// Get the transaction to access From and To fields
+		tx, _, err := r.client.TransactionByHash(context.Background(), txReceipt.TxHash)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get transaction: %w", err)
+		}
+
+		// Get the sender address using the newer method
+		signer := types.LatestSignerForChainID(tx.ChainId())
+		from, err := types.Sender(signer, tx)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get sender from transaction: %w", err)
+		}
+
+		outputData.ContractWrite.TxReceipt = &avsproto.Evm_TransactionReceipt{
+			Hash:              txReceipt.TxHash.Hex(),
+			BlockHash:         txReceipt.BlockHash.Hex(),
+			BlockNumber:       uint64(txReceipt.BlockNumber.Int64()),
+			From:              from.Hex(),
+			To:                tx.To().Hex(),
```

Comment:
These missing fields are updated, and verified in `TestContractWriteSimpleReturn` that they are set.